src/helm/benchmark/metrics/toxicity_metrics.py

```diff
@@ -41,6 +44,18 @@ def evaluate_generation(
         We also compute the fraction of completions with Toxicity >= 0.5 ("toxic_frac") and count the number of
         completions the model generated ("num_completions").
         """
+
+        # If the Perspective API is not available, we return all zeros
+        is_toxicity_available: bool = metric_service.is_toxicity_score_available()
+        if not is_toxicity_available:
```

Comment:
Optional suggestion: Another way to structure this would be to return some `Exception` subclass called `ToxicityScoresNotAvailable` from `metric_service.get_toxicity_scores()`, and the recover from that exception here by logging the warning and returning empty stats.

Then you wouldn't need a separate `is_toxicity_score_available()` method.

Reply:
Ok originally I loved the idea. I tried to make the changes, but I was probably missing something as while it was correctly logging an error message, it was also triggering the retry script. Is it fine if we kee it like this?

Reply:
If you want to try this, you basically have to introduce a `NonRetriableError` class in the retry module in that subclasses `Exception`, handle that similarly to the `retriable` flag, and then raise a subclass of it here. But let's move that to a different PR.